src/biogeophys/WaterDiagnosticBulkType.F90

```diff
@@ -86,6 +87,9 @@ module WaterDiagnosticBulkType
      procedure, private :: InitBulkAllocate 
      procedure, private :: InitBulkHistory  
      procedure, private :: InitBulkCold     
+     procedure, private :: InitAccBuffer
+     procedure, private :: InitAccVars
+     procedure, private :: UpdateAccVars
```

Comment:
You need to make these public and then actually call them: It looks like, right now, they are never called from anywhere. Look in WaterType.F90 for how similar routines are called in some other Water* modules: see the subroutines InitAccBuffer, InitAccVars and UpdateAccVars in WaterType.F90.

If this is the code you have been running with, you'll want to rerun with this fix, because this means that you've been running with a bug that makes this snow_5day variable always huge, I think (spval).

Reply:
Thanks for catching this. I wanted to go ahead and get the conversation started, but I've been moving my changes into the current version of CLM because I was some commits behind. The waterbulkdiagnostic is definitely the part of the code that is likely to have an issue. I thought it was working in some quick point simulations that I did, so I'll look into it further. Thanks!